charts/tekton-pipelines/values.yaml

```diff
@@ -60,13 +60,19 @@ buildpacks:
           resources: {}
           workingDir: $(resources.inputs.app.path)
           script: |
-            #!/bin/bash
-            [ -f package.json ] && (
-              echo "install node.js dependencies"
-              yes | npm install --silent;
-              npm run build:$(params.environment);
-              cp -rf {project.toml,nginx.conf,buildpack.yml,nginx.d,.nginx.d,nginx.*,httpd.conf,.http.d,httpd.d} $(params.source_subpath)  2>/dev/null;
-              chown -R $(params.user_id):$(params.group_id) $(params.source_subpath); ) || (echo "unable to build $(params.environment) env" && exit 1;)
+          #!/bin/bash
+          if [[ -f package.json ]]
+          then
+            (
+            echo "install node.js dependencies";
+            npm ci --silent &&\
+            npm run build:$(params.environment) &&\
+            cp -rf {project.toml,nginx.conf,buildpack.yml,nginx.d,.nginx.d,nginx.*,httpd.conf,.http.d,httpd.d} $(params.source_subpath)  2>/dev/null &&\
+            chown -R $(params.user_id):$(params.group_id) $(params.source_subpath)
+            ) || (echo "unable to build $(params.environment) env" && exit 1);
+          else
+            echo "no package.json"
+          fi
```

Comment:
let's make it cleaner:

see example:

```sh
#!/bin/bash
set -Eeo pipefail

[ -f package.json ] && {
  echo "install node.js dependencies"
  yes | npm install --silent
  npm run build:prod && rm -rf node_modules && echo "remove node_modules dir" || echo "unable to build prod" && exit 1
} || echo "No package.json found"

echo "done"
```

test with empty package.json

```
➜ ./test.sh         
install node.js dependencies
npm ERR! code EJSONPARSE
npm ERR! file /home/dmitry/package.json
npm ERR! JSON.parse Failed to parse json
npm ERR! JSON.parse Unexpected end of JSON input while parsing near ''
npm ERR! JSON.parse Failed to parse package.json data.
npm ERR! JSON.parse package.json must be actual JSON, not just JavaScript.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/dmitry/.npm/_logs/2022-02-10T23_21_43_863Z-debug.log
unable to build prod

✗  echo $?
1

```

just replace with `npm ci`. 

Also adjust the same for PHP step. As they also have npm install step. And adjust based on the code below.